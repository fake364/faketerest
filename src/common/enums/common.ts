export enum GENDER_ENUM {
  FEMALE = -1,
  NOT_SPECIFIED = 0,
  MALE
}
