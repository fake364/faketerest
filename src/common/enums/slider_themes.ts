export enum SLIDER_THEMES {
	DINNER_IDEA = 'dinnerIdea',
	HOME_DECOR = 'homeDecor',
	NEW_OUTFIT = 'newOutfit',
	THUMB_IDEA = 'thumbIdea'
}
