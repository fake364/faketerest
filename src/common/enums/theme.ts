export enum THEME_TYPE {
	BASE = 'base',
	BUSINESS = 'business',
	NEWSROOM = 'newsroom'
}