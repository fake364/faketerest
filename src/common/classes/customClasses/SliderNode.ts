import { SLIDER_THEMES } from '../../enums/slider_themes';

export const SLIDER_ARRAY = [
  SLIDER_THEMES.DINNER_IDEA,
  SLIDER_THEMES.HOME_DECOR,
  SLIDER_THEMES.NEW_OUTFIT,
  SLIDER_THEMES.THUMB_IDEA
];

export default class SliderNode {
  private readonly _type: SLIDER_THEMES;

  get type(): SLIDER_THEMES {
    return this._type;
  }

  constructor(type: SLIDER_THEMES) {
    this._type = type;
  }

  public static createThemesArray = (themes: SLIDER_THEMES[]) =>
    themes.map((theme) => new SliderNode(theme));
}
