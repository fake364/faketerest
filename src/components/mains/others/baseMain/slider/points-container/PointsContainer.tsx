import React from 'react';
import SliderPoint from '../slider-point/SliderPoint';
import SliderNode from '../../../../../../common/classes/customClasses/SliderNode';
import { SLIDER_THEMES } from '../../../../../../common/enums/slider_themes';

type Props = {
	sliderNodes: SliderNode[];
	shownType: SLIDER_THEMES;
};

const PointsContainer: React.FC<Props> = ({ sliderNodes, shownType }) => {
	return (
		<div className="flex justify-center mt-10">
			{sliderNodes.map(({ type: nodeType }) => (
				<SliderPoint
					pointType={nodeType}
					shownType={shownType}
					key={nodeType}
				/>
			))}
		</div>
	);
};

export default PointsContainer;
