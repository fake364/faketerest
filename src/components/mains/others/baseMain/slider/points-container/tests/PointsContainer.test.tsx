import React from 'react';
import { shallow } from 'enzyme';
import PointsContainer from '../PointsContainer';
import { SLIDER_THEMES } from '../../../../../../../common/enums/slider_themes';
import SliderNode from '../../../../../../../common/classes/customClasses/SliderNode';

describe('PointsContainer', () => {
  it('should render some slider nodes', () => {
    const wrapper = shallow(
      <PointsContainer
        sliderNodes={[
          new SliderNode(SLIDER_THEMES.HOME_DECOR),
          new SliderNode(SLIDER_THEMES.NEW_OUTFIT)
        ]}
        shownType={SLIDER_THEMES.HOME_DECOR}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
