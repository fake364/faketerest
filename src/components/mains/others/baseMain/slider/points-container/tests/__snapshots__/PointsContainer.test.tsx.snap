// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PointsContainer should render some slider nodes 1`] = `
<div
  className="flex justify-center mt-10"
>
  <SliderPoint
    key="homeDecor"
    pointType="homeDecor"
    shownType="homeDecor"
  />
  <SliderPoint
    key="newOutfit"
    pointType="newOutfit"
    shownType="homeDecor"
  />
</div>
`;
